src/lib/operators/aggregate_sort.cpp

```diff
@@ -186,46 +191,63 @@ void AggregateSort::_aggregate_values(const std::set<RowID>& group_boundaries, c
+ * @param current_secondary_aggregate the value of a supportive aggregate - used by StdDevSamp
+ * @param value_count the number of non-null values - used by COUNT(<name>), AVG and StdDevSamp
  * @param value_count_with_null the number of rows  - used by COUNT(*)
  * @param unique_value_count the number of unique values
  */
 template <typename AggregateType, AggregateFunction function>
 void AggregateSort::_set_and_write_aggregate_value(
     std::vector<AggregateType>& aggregate_results, std::vector<bool>& aggregate_null_values,
     const uint64_t aggregate_group_index, [[maybe_unused]] const uint64_t aggregate_index,
-    std::optional<AggregateType>& current_aggregate_value, [[maybe_unused]] const uint64_t value_count,
-    [[maybe_unused]] const uint64_t value_count_with_null, [[maybe_unused]] const uint64_t unique_value_count) const {
+    std::optional<AggregateType>& current_primary_aggregate, std::optional<AggregateType>& current_secondary_aggregate,
+    [[maybe_unused]] const uint64_t value_count, [[maybe_unused]] const uint64_t value_count_with_null,
+    [[maybe_unused]] const uint64_t unique_value_count) const {
   if constexpr (function == AggregateFunction::Count) {  // NOLINT
     if (this->_aggregates[aggregate_index].column) {
       // COUNT(<name>), so exclude null values
-      current_aggregate_value = value_count;
+      current_primary_aggregate = value_count;
     } else {
       // COUNT(*), so include null values
-      current_aggregate_value = value_count_with_null;
+      current_primary_aggregate = value_count_with_null;
     }
   }
   if constexpr (function == AggregateFunction::Avg && std::is_arithmetic_v<AggregateType>) {  // NOLINT
     // this ignores the case of Avg on strings, but we check in _on_execute() this does not happen
 
     if (value_count == 0) {
       // there are no non-null values, the average itself must be null (otherwise division by 0)
-      current_aggregate_value = std::optional<AggregateType>();
+      current_primary_aggregate = std::optional<AggregateType>();
     } else {
       // normal average calculation
-      current_aggregate_value = *current_aggregate_value / value_count;
+      current_primary_aggregate = *current_primary_aggregate / value_count;
+    }
+  }
+  if constexpr (function == AggregateFunction::StdDevSamp && std::is_arithmetic_v<AggregateType>) {  // NOLINT
+    // this ignores the case of StdDevSamp on strings, but we check in _on_execute() this does not happen
+
+    if (value_count <= 1) {
+      current_primary_aggregate = std::optional<AggregateType>();
+      current_secondary_aggregate = std::optional<AggregateType>();
+    } else {
+      // normal StdDevSamp calculation
+      const auto sum = *current_primary_aggregate;
+      const auto sum_square = *current_secondary_aggregate;
+      const double n = value_count;
+      const auto variance = (1 / (n - 1)) * (sum_square - (1 / n) * sum * sum);
+      current_primary_aggregate = std::sqrt(variance);
```

Comment:
This is the first aggregate functions that is a bit more complex which makes the redundancy of the computations of the aggregate functions in both aggregate operators quite obvious now.